### PR TITLE
feat: add useObserveChildren option

### PR DESCRIPTION
**Pull request by daybrush**
## Issue

#61

### CODE REVIEW

File: src/ContainerManager.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@
  */
 import Component from "@egjs/component";
 import { DestroyOptions, SizeRect } from "./types";
-import { ResizeWatcher } from "./ResizeWatcher";
+import { OnResizeWatcherResize, ResizeWatcher } from "./ResizeWatcher";
```
Comment (qpitlove) [inline src/ContainerManager.ts]: The word Resize seems to be repeated.
And what about camel case naming?
`onResizeWatcherObserve` or `onResizeWatcher`
https://vuejs.org/api/composition-api-lifecycle.html#onmounted

File: test/unit/Grid.spec.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import * as sinon from "sinon";
-import { MOUNT_STATE } from "../../src";
+import { MOUNT_STATE, OnRenderComplete } from "../../src";
```
Comment (qpitlove) [inline test/unit/Grid.spec.ts:2]: how about camecase like `onRenderComplete`

**Review by qpitlove — APPROVED** [review]
It would be nice if the Readme or api document included a statement that performance issues may occur when using the useObserveChildren option.

File: src/ContainerManager.ts (hunk 1 above)
Comment (WoodNeck) [reply src/ContainerManager.ts]: Or just `ResizeEvent` or something.
You know, namings like `OnXXX` look more proper for methods(functions) or attributes.
For interfaces, I think using nouns/adjectives is more proper than using verbs.
Like, `Observer` / `Observable` but not `Observe`

File: src/types.ts [hunk 3]
```diff
@@ -13,50 +13,137 @@ import { ItemRenderer, ItemRendererStatus } from "./ItemRenderer";
...
- * @property - The size of the outline. If the outline size is 0, it is calculated according to the grid type. (default: 0) <ko> outline의 사이즈. 만약 outline의 사이즈가 0이면, grid의 종류에 따라 계산이 된다. (default: 0) </ko>
- * @property - The default direction value when direction is not set in the render option. <ko>render옵션에서 direction을 미설정시의 기본 방향값.</ko>
- * @property - Whether to get the size as rounded size(offsetWidth, offsetHeight). Set to true if transform is applied to the container. If false, get the size through getBoundingClientRect. (default: true)
- <ko>사이즈를 반올림된 사이즈(offsetWidth, offsetHeight)로 가져올지 여부. container에 transform이 적용되어 있다면 true로 설정해라. false면 getBoundingClientRect를 통해 사이즈를 가져온다. (default: true)</ko>
- * @property - Whether to use ResizeObserver event to detect container size change when `autoResize` option is used. (default: false)<ko>autoResize 옵션 사용시 container의 사이즈 변화 감지를 위해 ResizeObserver 이벤트를 사용할지 여부. (default: false)</ko>
  * @property - You can set the ItemRenderer directly externally. <ko>외부에서 직접 ItemRenderer를 설정할 수 있다.</ko>
  * @property - You can set the ContainerManager directly externally. <ko>외부에서 직접 ContainerManager를 설정할 수 있다.</ko>
```
Comment (WoodNeck) [inline src/types.ts]: Why those are not removed?

**Review by WoodNeck — APPROVED** [review]
I think it would be fine with just `observeChildren`, but anyway
LGTM

**Review by malangfox — APPROVED** [review]
LGTM

**Pull request merged**